packages/ckeditor5-ui/src/editorui/editorui.ts

```diff
@@ -322,6 +324,34 @@ export default abstract class EditorUI extends /* #__PURE__ */ ObservableMixin()
 		return this._editableElementsMap;
 	}
 
+	/**
+	 * Initializes menu bar.
+	 */
+	protected _initMenuBar( menuBarView: MenuBarView ): void {
+		const menuBarViewElement = menuBarView.element!;
+
+		this.focusTracker.add( menuBarViewElement );
+		this.editor.keystrokes.listenTo( menuBarViewElement );
+
+		const normalizedMenuBarConfig = normalizeMenuBarConfig( this.editor.config.get( 'menuBar' ) || {} );
+
+		menuBarView.fillFromConfig( normalizedMenuBarConfig, this.componentFactory );
+
+		this.editor.keystrokes.set( 'Esc', ( data, cancel ) => {
+			if ( menuBarViewElement.contains( this.editor.ui.focusTracker.focusedElement ) ) {
+				this.editor.editing.view.focus();
+				cancel();
+			}
+		} );
+
+		this.editor.keystrokes.set( 'Alt+F9', ( data, cancel ) => {
+			if ( !menuBarViewElement.contains( this.editor.ui.focusTracker.focusedElement ) ) {
+				menuBarView!.focus();
+				cancel();
+			}
+		} );
```

Comment:
This handler has to be extended after merging this PR: https://github.com/ckeditor/ckeditor5/pull/16573, we added focus bordering flag assign on alt+f9.

Reply:
I see that you migrate it, thanks!